apps/api/src/cachereportfour.ts

```diff
@@ -0,0 +1,29 @@
+import { NestFactory } from '@nestjs/core';
+import { Context, Handler } from 'aws-lambda';
+import { AppModule } from './app.module';
+import { AppLogger } from './common/logger.service';
+import { ReportService } from './report/report.service';
+
+/**
+ * Design this function to trigger existing NestJs appliation services without Api-Getway
+ * All the schedule and backgroud job trigger will be added here.
+ * This handler will cache the report 4 data for every existing period
+ */
+export const handler: Handler = async (event, context: Context) => {
+  const app = await NestFactory.createApplicationContext(AppModule);
+  const reportService = app.get(ReportService);
+
+  const appLogger = app.get(AppLogger);
+  appLogger.log({ event });
+  appLogger.log({ context });
+  try {
+    if (event.path === 'cache-report-four') {
+      appLogger.log('Start caching report 4...');
+
+      await reportService.saveReportFourCache();
```

Comment:
How about changing `saveReportFourCache` to `updateReportCache`? We could add reports to that function later if we need to cache more reports without modifying the lambda.